src/convertToObject.js

```diff
@@ -7,6 +7,24 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const obj = {};
+
+  const strings = sourceString.split(';').filter(Boolean);
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy before each operation. Moreover, methods like `filter` don't mutate the original array, so you can safely chain another method after it without worrying about changing the original `sourceString`.